bin/setup

```diff
@@ -7,6 +7,71 @@ fancy_echo() {
   printf "\\n$fmt\\n" "$@"
 }
 
+install_linux_deps() {
+  local LINUX_DEPS="automake autoconf autogen libtool gcc curl"
+  local flavor=`grep "^ID=" /etc/os-release | cut -d"=" -f 2`
+  echo $flavor
+  case $flavor in
+    debian|ubuntu)
+      sudo apt-get install -y $LINUX_DEPS libgmp-dev
+      ;;
+    fedora|centos)
+      sudo yum install $LINUX_DEPS gmp-devel
+      ;;
+    *)
+      fancy_echo "Unrecognized distribution $flavor"
+      exit 1
+      ;;
+  esac
+}
+
+install_linux_rust() {
+  curl https://sh.rustup.rs -sSf | sh
+  if [ $? -ne 0 ]; then
+    fancy_echo "Rust installation failed"
+    exit 1
+  fi
+}
+
+install_rust() {
```

Comment:
`rustup` worked on my mac too. We can probably just use that everywhere and drop the `case` statement.